src/components/tables/DataPlanes/DialogFields.tsx

```diff
@@ -0,0 +1,226 @@
+                        sx={{
+                            color: 'text.secondary',
+                            fontSize: 12,
+                            lineHeight: 1,
+                        }}
+                    >
+                        {value}
+                    </TechnicalEmphasis>
+                    <CopyIconIndicator
+                        isCopied={isCopied}
+                        isHovered={isHovered}
+                    />
+                </Stack>
+            ) : (
+                <Typography color="text.secondary">{value ?? '-'}</Typography>
+            )}
+        </Stack>
+    );
+}
+
+export function ToggleField({
+    label,
+    options,
+    lowercaseButton,
+}: ToggleFieldProps) {
+    const { isCopied, handleCopy, setIsCopied } = useCopyToClipboard();
+    const [isHovered, setIsHovered] = useState(false);
+    const [isToggleHovered, setIsToggleHovered] = useState(false);
+
+    const validOptions = options.filter((opt) => Boolean(opt.value));
+
+    const [selectedKey, setSelectedKey] = useState<string>(
+        validOptions[0]?.key ?? ''
+    );
+
+    const handleSelectionChange = (
+        _event: React.MouseEvent<HTMLElement>,
+        newKey: string | null
+    ) => {
+        if (newKey !== null) {
+            setSelectedKey(newKey);
+            setIsCopied(false);
+        }
+    };
+
+    const currentOption = validOptions.find((opt) => opt.key === selectedKey);
+    const currentValue = currentOption?.value ?? null;
+
+    if (validOptions.length === 0) {
+        return null;
+    }
+
+    return (
+        <Stack
+            sx={{
+                py: 1,
+                cursor: currentValue ? 'pointer' : 'default',
+            }}
+            onClick={() => handleCopy(currentValue)}
+            onMouseEnter={() => setIsHovered(true)}
+            onMouseLeave={() => setIsHovered(false)}
+        >
+            <Stack
+                direction="row"
+                alignItems="center"
+                spacing={1}
+                sx={{
+                    mb: 1,
+                }}
+            >
+                <Typography variant="subtitle2" fontWeight={600}>
+                    {label}
+                </Typography>
+                <ToggleButtonGroup
+                    value={selectedKey}
+                    exclusive
+                    onChange={handleSelectionChange}
+                    // prevent the ToggleButtonGroup from triggering copy on click
+                    onClick={(e) => e.stopPropagation()}
+                    onMouseEnter={() => setIsToggleHovered(true)}
+                    onMouseLeave={() => setIsToggleHovered(false)}
+                    size="small"
+                >
+                    {validOptions.map((opt) => (
+                        <ToggleButton
+                            key={opt.key}
+                            value={opt.key}
+                            sx={{ py: 0.25, px: 1, borderRadius: 3 }}
+                        >
```

Comment:
We have `src/components/shared/buttons/OutlinedToggleButton.tsx` that can handle rendering the options.